src/ssd/ssd-titlebar.c

```diff
@@ -624,22 +605,11 @@ ssd_update_window_icon(struct ssd *ssd)
 			break;
 		}
 
-		/* Replace all the buffers in the button with the window icon */
 		struct ssd_button *button = node_ssd_button_from_node(part->node);
-		for (uint8_t state_set = 0; state_set <= LAB_BS_ALL; state_set++) {
-			struct wlr_scene_node *node = button->nodes[state_set];
-			if (node) {
-				struct scaled_img_buffer *img_buffer =
-					scaled_img_buffer_from_node(node);
-				scaled_img_buffer_update(img_buffer, icon_img,
-					theme->window_button_width,
-					theme->window_button_height,
-					icon_padding);
-			}
-		}
+		struct scaled_icon_buffer *icon_buffer =
+			scaled_icon_buffer_from_node(button->nodes[0]);
+		scaled_icon_buffer_set_app_id(icon_buffer, app_id);
```

Comment:
Is it ensure that `nodes[0]` is always available? If yes I think an `assert(icon_buffer)` and potentially a short comment explaining why its always available would be good here.